shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1635,22 +1624,20 @@ - (CGRect)firstRectForRange:(UITextRange*)range {
 }
 
 - (CGRect)caretRectForPosition:(UITextPosition*)position {
-  // TODO(cbracken) Implement.
-
-  // As of iOS 14.4, this call is used by iOS's
-  // _UIKeyboardTextSelectionController to determine the position
-  // of the floating cursor when the user force touches the space
-  // bar to initiate floating cursor.
-  //
-  // It is recommended to return a value that's roughly the
-  // center of kSpacePanBounds to make sure the floating cursor
-  // has ample space in all directions and does not hit kSpacePanBounds.
-  // See the comments in beginFloatingCursorAtPoint.
-  return CGRectZero;
-}
-
-- (CGRect)bounds {
-  return _isFloatingCursorActive ? kSpacePanBounds : super.bounds;
+  NSInteger index = ((FlutterTextPosition*)position).index;
+  NSArray<UITextSelectionRect*>* rects =
+      [self selectionRectsForRange:[FlutterTextRange
+                                       rangeWithNSRange:fml::RangeForCharactersInRange(
+                                                            self.text,
+                                                            NSMakeRange(MAX(0, index - 1), 1))]];
+  if (rects.count == 0) {
+    return CGRectZero;
+  }
+  if (index == 0) {
+    return CGRectMake(rects[0].rect.origin.x, rects[0].rect.origin.y, 0, rects[0].rect.size.height);
+  }
+  return CGRectMake(rects[0].rect.origin.x + rects[0].rect.size.width, rects[0].rect.origin.y, 0,
+                    rects[0].rect.size.height);
```

Comment:
Might be easier to reach if `rects[0].rect` was in a local.  They could also use comments as to why and in what way the rects are being adjusted.